src/main/java/net/sf/javaanpr/intelligence/Parser.java

```diff
@@ -161,30 +161,31 @@ public void invertFlags() {
     /**
      * Syntactically parses text from the given {@link RecognizedPlate} in the specified analysis mode.
      * <p>
-     * Syntax analysis mode: <ul>
-     * <li>0 (do not parse)</li>
-     * <li>1 (only equal length)</li>
-     * <li>2 (equal or shorter)</li>
-     * </ul>
      *
      * @param recognizedPlate the plate to parse
      * @param syntaxAnalysisMode the mode in which to parse
      * @return the parsed recognized plate text
      */
-    public String parse(RecognizedPlate recognizedPlate, int syntaxAnalysisMode) { // TODO enum
-        if (syntaxAnalysisMode == 0) {
-            Main.rg.insertText(
-                    " result : " + recognizedPlate.getString() + " --> <font size=15>" + recognizedPlate.getString()
-                            + "</font><hr><br>");
-            return recognizedPlate.getString();
-        }
-
+    public String parse(RecognizedPlate recognizedPlate, SyntaxAnalysisMode syntaxAnalysisMode) {
```

Comment:
This changes one of the methods in the "public API" (there is no such thing currently, but I suspect people use the method in their code). We should be able to do this change for the next major release, but we should document it somehow. I added the following file: `docs/upgradeRecipes/upgradeRecipe-2.0.0.adoc`. It will be of great help to all users if they wish to upgrade in the future.

Just add something along the lines of
```
=== `Parser.parse` method parameters

The method `String Parser.parse(RecognizedPlate, int)` changed to
`String Parser.parse(RecognizedPlate, SyntaxAnalysisMode)`.

Workaround: `parse(plate, syntaxAnalysisModeInt)`
-> `parse(plate, SyntaxAnalysisMode.getSyntaxAnalysisModeFromInt(syntaxAnalysisModeInt))`
```

To the "Minor API changes" part of that file.